src/actions/search-results.js

```diff
@@ -164,6 +167,7 @@ export function mixDataInput () {
 
 export function startSearch () {
   return (dispatch, getState) => {
+    mixer = mixDataInput();
```

Comment:
Create a new mixer per search, otherwise the counts and buffer arrays are not reset and can potentially contain results from the previous search.